core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala

```diff
@@ -781,7 +783,9 @@ class SnappyStoreHiveCatalog(externalCatalog: SnappyExternalCatalog,
           // TODO: the authorizationID should be correctly set in SparkSQLExecuteImpl
           // using LCC.getAuthorizationId() itself rather than getUserName()
           val user = snappySession.conf.get(Attribute.USERNAME_ATTR, "")
-          if (user.nonEmpty) {
+          if (user.nonEmpty && !(
+              tableIdent.schemaName.equalsIgnoreCase(SchemaDescriptor.IBM_SYSTEM_SCHEMA_NAME )
+              && tableIdent.table.equalsIgnoreCase(SnappyStoreHiveCatalog.dummyTableName))) {
```

Comment:
This will allow all users to write to SYSIBM.DUMMY1 table? I hope Derby layer restricts this somewhere.

Reply:
I also hope. Bcoz I am also not comfortable with the change. But I have no idea what else can be done at this point to fix the issue & why it started coming now.

Reply:
Confirmed that no changes to SYSIBM.DUMMY1 are allowed and is restricted in GemXD layer in any case.

The reason this is coming is because catalog's registerDataSourceTable() now has a checkSchemaPermission call to restrict CREATE EXTERNAL TABLE, VIEW etc as per schema permissions that do not hit storage layers rather are only in hive catalog. Looks like GRANT/REVOKE also are calling registerDataSourceTable() with the dummy table. This should not happen in any case and registerDataSourceTable can return immediately for SYSIBM.DUMMY1 table. Perhaps that is a better fix but can be tracked for later if above changes have already been tested.